src/zarr/abc/store.py

```diff
@@ -371,6 +371,36 @@ def list_dir(self, prefix: str) -> AsyncGenerator[str, None]:
         """
         ...
 
+    async def delete_dir(self, prefix: str, recursive: bool = True) -> None:
+        """
+        Remove all keys and prefixes in the store that begin with a given prefix.
```

Comment:
I think it would make life much easier for `Store` implementers if we declare that `prefix` will always be the path of a group or a array. Otherwise, all implementations need to deal with unnecessary edge cases like `delete_dir("path/to/array/c/0")` or worse.

[Icechunk](https://icechunk.io/) is an example of a store that could very cleanly and efficiently implement the case for a group/array prefix, but it would struggle to handle other prefixes.

Reply:
What is the use case for `recursive = False` ? If zarr itself doesn't have one, I'd remove the argument.

Reply:
I can see a use case for `delete_dir("path/to/array/c/")` (remove all chunks from an array) so I think we should allow that -- even if it makes life a little harder in icechunk.

Reply:
That's totally fine. Let's document the three possible prefix cases, and say that stores are allowed to do undefined behavior for other prefixes